inc/functions/files.php

```diff
@@ -1470,3 +1470,32 @@ function _rocket_is_windows_fs( $hard_reset = false ) { // phpcs:ignore WordPres
 function _rocket_get_wp_rocket_cache_path() { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedFunctionFound
 	return _rocket_normalize_path( rocket_get_constant( 'WP_ROCKET_CACHE_PATH' ) );
 }
+
+/**
+ * Get an array of php files in directory.
+ *
+ * @param string $dir_path Full path to the directory.
+ *
+ * @return array php files in the directory.
+ *               [...SplFileInfo file]
+ *
+ * @since  ver 3.6.1
+ *
+ * @author Caspar Green
+ */
+function _rocket_get_directory_php_files_array( $dir_path ) { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedFunctionFound
+	try {
+		$config_dir = new FilesystemIterator( (string) $dir_path );
+	} catch ( Exception $e ) {
+		return [];
+	}
+	$files = [];
+
+	foreach ( $config_dir as $file ) {
+		if ( $file->isFile() && 'php' === $file->getExtension() ) {
```

Comment:
@iCaspar do you think using `RegexIterator` to target only `.php` files would be more efficient and/or faster? Curious of your thoughts.

Reply:
May I reply here?
at another PR #2764 I added generic function that may be used here.
https://github.com/wp-media/wp-rocket/blob/0833d81e74efa7ea30a255f5b4adb878221779b0/inc/functions/files.php#L1320-L1344
Your thoughts?

Reply:
I do like this `IteratorIterator` approach. Performance-wise, the difference is probably negligible compared to the time it will take to access the filesystem (depending on the kind of filesystem). But the versatility of using regex is far more reusable. :)

Question. Why two separate try/catch blocks? If you're going to bail out with the empty array in either case, can you put it all into a single block and catch it in one place? Would read more easily, I think. :)

Reply:
@engahmeds3ed Let's move this conversation to your PR.